lua/ui/game/painting/PaintingCanvas.lua

```diff
@@ -0,0 +1,332 @@
+--******************************************************************************************************
+--** Copyright (c) 2025 Willem 'Jip' Wijnia
+--**
+--** Permission is hereby granted, free of charge, to any person obtaining a copy
+--** of this software and associated documentation files (the "Software"), to deal
+--** in the Software without restriction, including without limitation the rights
+--** to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+--** copies of the Software, and to permit persons to whom the Software is
+--** furnished to do so, subject to the following conditions:
+--**
+--** The above copyright notice and this permission notice shall be included in all
+--** copies or substantial portions of the Software.
+--**
+--** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+--** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+--** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+--** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+--** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+--** OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+--** SOFTWARE.
+--******************************************************************************************************
+
+local LayoutFor = import('/lua/maui/layouthelpers.lua').ReusedLayoutFor
+local Bitmap = import('/lua/maui/bitmap.lua').Bitmap
+
+local GetPaintingCanvasAdapter = import('/lua/ui/game/painting/ShareAdapters/PaintingCanvasAdapterFactory.lua').GetPaintingCanvasAdapter
+local CreatePainting = import("/lua/ui/game/painting/Painting.lua").CreatePainting
+
+local DebugComponent = import("/lua/shared/components/DebugComponent.lua").DebugComponent
+
+local PaintingCanvasInstances = TrashBag()
+
+local SyncIdentifier = "PaintingCanvas.lua"
+
+---@alias UIPaintingCanvasActiveInteraction 'Create' | 'Delete' | 'Mute'
+
+--- Responsible for glueing together the painting, the brush stroke and the painting network adapter.
+---@class UIPaintingCanvas : Bitmap, DebugComponent, Renderable, Destroyable
+---@field Adapter UIPaintingCanvasAdapter
+---@field Painting UIPainting
+---@field Trash TrashBag
+---@field WorldView WorldView                   # Worldview that this canvas is for
+---@field InhibitionSet table<string, boolean>  # A set of reasons for the canvas to be disabled.
+---@field PaintingBrush UIPaintingBrush
+PaintingCanvas = Class(Bitmap, DebugComponent) {
+
+    ---@param self UIPaintingCanvas
+    ---@param worldview WorldView
+    __init = function(self, worldview)
+        Bitmap.__init(self, worldview)
+
+        self.Trash = TrashBag()
+        self.Painting = CreatePainting()
+        self.Adapter = self.Trash:Add(GetPaintingCanvasAdapter(self))
+
+        self.WorldView = worldview
+
+        -- register us to render
+        self.WorldView:RegisterRenderable(self, tostring(self))
```

Comment:
Update to match refactor
```suggestion
        self.WorldView:AddShape(self, tostring(self))
```